qstash/api/api-ratelimiting.mdx

```diff
@@ -0,0 +1,68 @@
+---
+title: "API Rate Limit Response"
+description: "This page documents the rate limiting behavior of our API and explains how to handle different types of rate limit errors."
+---
+
+## Overview
+
+Our API implements rate limiting to ensure fair usage and maintain service stability. There are three types of rate limits:
+
+1. **Daily rate limit**
+2. **Burst rate limit**
+3. **Chat-based rate limit**
+
+When a rate limit is exceeded, the API returns a 429 status code along with specific headers that provide information about the limit, remaining requests/tokens, and reset time.
+
+### Daily Rate Limit
+
+This is a **daily** limit applied across all API endpoints.
+
+**Headers**:
+
+- `RateLimit-Limit`: Maximum number of requests allowed per day
+- `RateLimit-Remaining`: Remaining number of requests for the day
+- `RateLimit-Reset`: Time (in unix timestamp) when the daily limit will reset 
+
+### Burst Rate Limit
+
+This is a short-term limit **per second** to prevent rapid bursts of requests.
+
+**Headers**:
+
+- `Burst-RateLimit-Limit`: Maximum number of requests allowed in the burst window (1 second)
+- `Burst-RateLimit-Remaining`: Remaining number of requests in the burst window (1 second)
+- `Burst-RateLimit-Reset`: Time (in unix timestamp) when the burst limit will reset
+
+### Chat-based Rate Limit
+
+This limit is applied to chat-related API endpoints.
+
+**Headers**:
+
+- `x-ratelimit-limit-requests`: Maximum number of requests allowed
+- `x-ratelimit-limit-tokens`: Maximum number of tokens allowed
+- `x-ratelimit-remaining-requests`: Remaining number of requests
+- `x-ratelimit-remaining-tokens`: Remaining number of tokens
+- `x-ratelimit-reset-requests`: Time (in unix timestamp) until the request limit resets
+- `x-ratelimit-reset-tokens`: Time (in unix timestamp) when the token limit will reset
+
+### Example Rate Limit Error Handling
+
+
+```typescript Handling in TypeScript 
+import { QstashChatRatelimitError, QstashDailyRatelimitError, QstashRatelimitError } from "@upstash/qstash"
+
+try {
+  const result = await client.events();
+} catch (error) {
+  if (error instanceof QstashChatRatelimitError) {
+    console.log('Chat rate limit exceeded. Retry after:', error.resetRequests);
+  } else if (error instanceof QstashDailyRatelimitError) {
+    console.log('Daily rate limit exceeded. Retry after:', error.reset);
+  } else if (error instanceof QstashRatelimitError) {
+    console.log('Burst rate limit exceeded. Retry after:', error.reset);
+  } else {
+    console.error('An unexpected error occurred:', error);
+  }
```

Comment:
Maybe we should put different API request for each error type? Because `client.events()` never throws `QstashChatRatelimitError`.